c2rust-analyze/src/rewrite/expr/mir_op.rs

```diff
@@ -128,8 +135,52 @@ impl<'a, 'tcx> ExprRewriteVisitor<'a, 'tcx> {
         match stmt.kind {
             StatementKind::Assign(ref x) => {
                 let (pl, ref rv) = **x;
-                let pl_ty = self.acx.type_of(pl);
-                self.enter_assign_rvalue(|v| v.visit_rvalue(rv, pl_ty));
+                let pl_lty = self.acx.type_of(pl);
+
+                if pl.is_indirect() && self.acx.local_tys[pl.local].ty.is_any_ptr() {
+                    let local_addr = self.acx.local_tys[pl.local].label;
+                    let perms = self.perms[local_addr];
+                    let flags = self.flags[local_addr];
+                    if let (Ownership::Cell, _) = type_desc::perms_to_desc(perms, flags) {
+                        // this is an assignment like `let x = 2` but `x` has CELL permissions
```

Comment:
`let` seems inappropriate here if this assignment corresponds to `CellSet` and not `CellNew`.

Reply:
Agreed, fixed in c7a8216b